File: 프로그래머스 python/시저 암호.py
# ord() : 아스키코드로 변경 , chr() : 아스키 코드를 string  print(chr(ord(s[0])+n))
def solution(s, n):
    answer = []
    lst = list(''.join(s))
    print(lst)

    for i in lst:
        if i == ' ':
            answer.append(i)
        elif ord(i) >= 65 and ord(i) <= 90:  # 소문자
            if ord(i) + n > 90:  # Z값을 넘으면
                Temp = 90 - ord(i) + n
                answer.append(chr(64 + temp))
            else:
                answer.append(chr(ord(i) + n))
        elif ord(i) >= 97 and ord(i) <= 122:  # 대문자
            if ord(i) + n > 122:  # z값을 넘으면
                temp = 122 - ord(i) + n
                answer.append(chr(96 + temp))
            else:
                answer.append(chr(ord(i) + n))

    return ''.join(answer)

File: 프로그래머스 Javascript/시저 암호.js
function solution(s, n) {
  //문자열.charCodeAt(문자열 자릿수) /String.fromCharCode(아스키코드 번호)
  var answer = [];
  let stringArr = s.split("");
  let temp = 0;

  stringArr.forEach((item, index) => {
    if (item === " ") {
      answer.push(item);
    } else if (item.charCodeAt() >= 65 && item.charCodeAt() <= 90) {
      //대문자 범위
      if (item.charCodeAt() + n > 90) {
        //대문자 범위를 넘어가면
        temp = item.charCodeAt() + n - 90;
        answer.push(String.fromCharCode(64 + temp));
      } else {
        answer.push(String.fromCharCode(item.charCodeAt() + n));
      }
    } else if (item.charCodeAt() >= 97 && item.charCodeAt() <= 122) {
      // 소문자 범위
      if (item.charCodeAt() + n > 122) {
        //소문자 범위를 넘어가면
        temp = item.charCodeAt() + n - 122;
        answer.push(String.fromCharCode(96 + temp));
      } else {
        answer.push(String.fromCharCode(item.charCodeAt() + n));
      }
    }
  });
  return answer.join("");
}